src/libImaging/Jpeg2KDecode.c

```diff
@@ -742,8 +742,8 @@ j2k_decode_entry(Imaging im, ImagingCodecState state)
            swapped), bail. */
         if (tile_info.x0 >= tile_info.x1
             || tile_info.y0 >= tile_info.y1
-            || tile_info.x0 < image->x0
-            || tile_info.y0 < image->y0
+            || tile_info.x0 < (OPJ_INT32)image->x0
+            || tile_info.y0 < (OPJ_INT32)image->y0
```

Comment:
`x0` and `y0` are [signed, as set by OpenJPEG](https://github.com/uclouvain/openjpeg/blob/cbee7891a0ee664dd83ca09553d2e30da716a883/src/lib/openjp2/openjpeg.h#L1515). So the tile offset can be theoretically negative.
Part of this block is to check '[if the tile is outside the image area](https://github.com/python-pillow/Pillow/blob/0defffaece306ffa3ac910867c692fe94bfaddbc/src/libImaging/Jpeg2KDecode.c#L740)'. If the tile offset is negative, that is definitely outside the image area.

Reply:
Would this cause issues for images with `x0` or `y0` greater than 2**31, since [image->x0 is unsigned](https://github.com/uclouvain/openjpeg/blob/cbee7891a0ee664dd83ca09553d2e30da716a883/src/lib/openjp2/openjpeg.h#L680-L683)? The following suggestion would catch both cases:

```suggestion
            || tile_info.x0 < 0
            || tile_info.y0 < 0
            || (OPJ_UINT32)tile_info.x0 < image->x0
            || (OPJ_UINT32)tile_info.y0 < image->y0
```

One could argue that such an image could have no valid tiles, but I do not see another check specifically for that case.